File: lib/features/pay/domain/vpa.dart
class Vpa {
  final String address;
  final String verifiedName;

  const Vpa({
    required this.address,
    required this.verifiedName,
  });

  factory Vpa.fromJson(Map<String, dynamic> json) {
    return Vpa(
      address: json['address'] as String,
      verifiedName: json['verifiedName'] as String,
    );
  }
}

File: backend/routes/vpa.js
const express = require('express');
const { vpas } = require('../data/seed');
const { authenticateToken } = require('../middleware/auth');

const router = express.Router();

router.use(authenticateToken);

router.get('/:address', (req, res) => {
  const address = req.params.address.trim().toLowerCase();
  const vpa = vpas.find(v => v.address === address);

  if (!vpa || !vpa.verifiedName) {
    return res.status(404).json({
      error: { code: 'VPA_NOT_FOUND', message: 'No account found for this UPI ID' }
    });
  }

  res.json({
    address: vpa.address,
    verifiedName: vpa.verifiedName,
    bankName: vpa.bankName
  });
});

module.exports = router;
